tests/unit/Core/System/CustomEntity/Schema/SchemaUpdaterTest.php

```diff
@@ -186,6 +187,146 @@ public static function associationPairsProvider(): \Generator
         ];
     }
 
+    /**
+     * @param list<array{name: string, fields: string}> $entities
+     * @param array<string, list<string>> $notExpectedSchema
+     * @param array<string, list<string>> $expectedNonExistTableNames
+     */
+    #[DataProvider('associationWithIgnoreMissingReferencePairsProvider')]
+    public function testAssociationsWithIgnoreMissingReference(array $entities, array $notExpectedSchema, array $expectedNonExistTableNames): void
+    {
+        $schema = new Schema();
+
+        $updater = new SchemaUpdater();
+        $updater->applyCustomEntities($schema, $entities);
+
+        /** @var string $nonExistTableName */
```

Comment:
please do not use var annotations. Especially in tests you can simply assert the type. also see https://github.com/shopware/shopware/blob/trunk/coding-guidelines/core/writing-code-for-static-analysis.md